text/0000-derefered_composite_types.md

```diff
@@ -0,0 +1,165 @@
+- Feature Name: `derefered_composite_types`
+- Start Date: 2023-04-06
+- RFC PR: [rust-lang/rfcs#0000](https://github.com/rust-lang/rfcs/pull/0000)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+
+# Summary
+[summary]: #summary
+
+The proposal of Dereferenced (or Derefered) Composite Types is a Rust unification of Referenced Types and Dereferenced Types.
+
+Dereferenced Composite Types could help also to get rid of constructor boilerplate.
+
+The symbol `*` before the type name is a marker of Dereferenced Composite Type. 
+
+The new operator `=&` (and `=&&`, `=&&&`, ...) as a compound of borrow and assignment (`x = &2` same as `x =& 2`) is also required.
+
+
+# Motivation
+[motivation]: #motivation
+
+Currently, Rust has 
+- (1) dereferenced primitive types (like `i32`)
+- (2) referenced primitive types (like `&i32`) 
+- (3) referenced composite types (like `Box<i32>`)
+
+But there are no any dereferenced composite types.
+
+This is not universal. We wish to improve this. So we propose types like `*Box<i32>`.
```

Comment:
Could you expand this section? I have a hard time understanding the use cases for this feature. All of the code examples in this RFC are only a couple of lines long and they do not communicate what this feature is intended for. I would recommend using a real-world example that inspired you to create this RFC to show how this feature should be used.
The motivation "This model has a gap, it should be closed." is in my opinion not sufficient, such a feature also needs to cater to some real use cases and it needs to improve upon existing solutions (e.g. Macros, traits etc.) in order to be a useful addition.

Reply:
The only real advantages in real-world I found for Dereferenced Types are smart constructors. 
But I also found, that smart constructors could be implemented without Dereferenced Types. 
It is discussed in my [comment](https://github.com/rust-lang/rfcs/pull/3408#discussion_r1161094008)

Reply:
Then I do not really see the point of this RFC. Adding the syntax, types and desugaring just for nothing? It does not make sense.

I prefer my constructors explicit, since then you can immediately tell that you are not handling a primitive type. IIUC, your proposal still requires the type to be named somewhere:
```rust
let u: *Box<i32> = 5;
// vs
let u = Box::new(5);
```
I prefer the latter by a lot. And I do not like that the former does not interact with other Rust features (I cannot put just `5` into a function that takes a `Box<i32>`).